src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/TagHelperComponentManager.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    public class TagHelperComponentManager : ITagHelperComponentManager
+    {
+        private List<ITagHelperComponent> _tagHelperComponents;
+
+        public TagHelperComponentManager(IEnumerable<ITagHelperComponent> tagHelperComponents)
+        {
+            if (tagHelperComponents == null)
+            {
+                throw new ArgumentNullException(nameof(tagHelperComponents));
+            }
+
+            _tagHelperComponents = new List<ITagHelperComponent>(tagHelperComponents);
+        }
+
+        public IEnumerable<ITagHelperComponent> Components => _tagHelperComponents;
+
+        public void Add(ITagHelperComponent tagHelperComponent)
+        {
+            _tagHelperComponents.Add(tagHelperComponent);
+        }
```

Comment:
Why not `IList<>` for the components? Is is not supported to remove components?

I can see two sensible ways that `ITagHelperManager` deals with ordering in a consistent way.

1. Expose `IList<>` for the components. The components from DI are always put in order for you. Callers can add **or** insert a component wherever they want, reorder the list, remove stuff, etc.

2. Expose `IEnumerable<>` **or** `ICollection<>`. Calling add should find the right place in the list for the new component based on its order and insert it. Use `ICollection<>` if you want to support removing things or `IEnumerable<>` if you don't.